src/rgw/rgw_appmain.cc

```diff
@@ -468,24 +460,25 @@ int rgw::AppMain::init_frontends2(RGWLib* rgwlib)
   }
 
   std::string daemon_type = (nfs) ? "rgw-nfs" : "rgw";
-  r = store->register_to_service_map(dpp, daemon_type, service_map_meta);
+  r = env.store->register_to_service_map(dpp, daemon_type, service_map_meta);
   if (r < 0) {
     derr << "ERROR: failed to register to service map: " << cpp_strerror(-r) << dendl;
     /* ignore error */
   }
 
-  if (store->get_name() == "rados") {
+  if (env.store->get_name() == "rados") {
     // add a watcher to respond to realm configuration changes
-    pusher = std::make_unique<RGWPeriodPusher>(dpp, store, null_yield);
-    fe_pauser = std::make_unique<RGWFrontendPauser>(fes, *(implicit_tenant_context.get()), pusher.get());
+    pusher = std::make_unique<RGWPeriodPusher>(dpp, env.store, null_yield);
+    fe_pauser = std::make_unique<RGWFrontendPauser>(fes, pusher.get());
     rgw_pauser = std::make_unique<RGWPauser>();
     rgw_pauser->add_pauser(fe_pauser.get());
     if (lua_background) {
```

Comment:
since we store the lua background pointer inside `RGWProcessEnv` which is already a member of `AppMain`, maybe we don't need to also store it in `AppMain`, and use the `unique_ptr` in the env?

Reply:
@cbodley ?